modules/@apostrophecms/schema/index.js

```diff
@@ -687,180 +687,37 @@ module.exports = {
+        let manager = self.apos.doc.getManager(field.withType);
         if (!manager) {
           throw Error('join with type ' + field.withType + ' unrecognized');
         }
-        if (_.has(data, field.name)) {
-          let titleOrId = self.apos.launder.string(data[field.name]);
-          let criteria = {
-            $or: [
-              { titleSortified: self.apos.util.sortify(titleOrId) },
-              { _id: titleOrId }
-            ]
-          };
-          const result = await manager.find(req, criteria, { _id: 1 }).joins(false).published(null).toObject();
-          if (result) {
-            object[field.idField] = result._id;
-          } else {
-            delete object[field.idField];
-          }
-        } else if (_.has(data, field.idField)) {
-          object[field.idField] = self.apos.launder.id(data[field.idField]);
-        }
-      },
-      join: async function (req, field, objects, options) {
-        return self.joinDriver(req, joinr.byOne, false, objects, field.idField, undefined, field.name, options);
-      },
-      addQueryBuilder(field, query) {
-        // for joinByOne only the "OR" case makes sense
-        query.addBuilder(field.name, {
-          finalize: function () {
-            if (!self.queryBuilderInterested(query, field.name)) {
-              return;
-            }
-            const value = query.get(field.name);
-            const criteria = {};
-            // Even programmers appreciate shortcuts, so it's not enough that the
-            // sanitizer (which doesn't apply to programmatic use) accepts these
-            if (Array.isArray(value)) {
-              criteria[field.idField] = { $in: value };
-            } else if (value === 'none') {
-              criteria.$or = [];
-              let clause = {};
-              clause[field.idField] = null;
-              criteria.$or.push(clause);
-              clause = {};
-              clause[field.idField] = { $exists: 0 };
-              criteria.$or.push(clause);
-            } else {
-              criteria[field.idField] = value;
-            }
-            query.and(criteria);
-          },
-          choices: self.joinQueryBuilderChoices(field, query, '_id'),
-          launder: joinQueryBuilderLaunder
-        });
 
-        self.addJoinSlugQueryBuilder(field, query, '');
-      },
-      validate: function (field, options, warn, fail) {
-        if (!field.name.match(/^_/)) {
-          warn('Name of join field does not start with _. This is permitted for bc but it will fill your database with duplicate outdated data. Please fix it.');
-        }
-        if (!field.idField) {
-          if (field.idsField) {
-            fail('joinByOne takes idField, not idsField. You can also omit it, in which case a reasonable value is supplied.');
+        if (_.has(data, field.name)) {
+          if (field.min && field.min > data[field.name].length) {
+            throw new Error(`Minimum ${field.withType} required not reached.`);
```

Comment:
Look at how errors are thrown in other validators for schemas:

`throw self.apos.error('min');`

Reply:
I must have copied the wrong one with `throw new Error()` but for the record there are several like this in this file.